app/controllers/api/v1/recordings_controller.rb

```diff
@@ -57,19 +57,28 @@ def resync
         render_json(status: :ok)
       end
 
-      def publish_recording
-        publish = params[:publish]
+      # POST /api/v1/recordings/update_visibility.json
+      def update_visibility
         record_id = params[:record_id]
-        case publish
-        when 'true'
-          visibility = 'Published'
-        when 'false'
-          visibility = 'Unpublished'
+        visibility = params[:visibility]
```

Comment:
Agreed.

Reply:
This will have to be done when refactoring as I would have to make big changes to the RecordingController to allow the scoping.

Reply:
It may be a good idea even for consistency if we further catch invalid/missing params.
`return render_json status: :bad_request unless params[:visibility]`
until we do the refactoring.

Reply:
What validations?
The `ActiveRecord::Validations`?